doc/source/whatsnew/v0.19.2.txt

```diff
@@ -32,6 +32,7 @@ Bug Fixes
 
 
 
+- Bug in pd.read_csv - catch missing columns if usecols and header lengths match (:issue:`14671`)
```

Comment:
1) `pd.read_csv()` instead of `pd.read_csv`
2) Let's generalize this a little.  This PR is not actually about when `header` and `usecols` length match.  It's about properly handling situations when `usecols` provides non-existent columns.